integration_tests/benchmarks/object-detection/benchmark_script.py

```diff
@@ -233,60 +171,239 @@ def run_detailed_pr_curve_evaluation(dset: Dataset, model: Model):
+            except (
+                DatasetAlreadyExistsError,
+                ModelAlreadyExistsError,
+            ) as e:
+                client.delete_dataset("coco-raster")
+                client.delete_model("yolo-raster")
+                raise e
+
+            # gt raster ingestion
+            gt_raster_ingest_time = time_it(
+                ingest_groundtruths,
+                dataset=dset_raster,
+                path=current_directory / Path(gt_raster_filename),
+                limit=limit,
+                chunk_size=100,
+            )
 
-        try:
-            eval_ = run_base_evaluation(dset=dset, model=model)
-        except TimeoutError:
-            raise TimeoutError(
-                f"Evaluation timed out when processing {limit} datums."
+            # gt raster finalization
+            gt_raster_finalization_time = time_it(dset_raster.finalize)
+
+            # pd raster ingestion
+            raster_datum_uids = [
+                datum.uid for datum in dset_raster.get_datums()
+            ]
+            pd_raster_ingest_time = time_it(
+                ingest_predictions,
+                dataset=dset_raster,
+                model=model_raster,
+                datum_uids=raster_datum_uids,
+                path=current_directory / Path(pd_raster_filename),
+                limit=limit,
+                chunk_size=100,
             )
 
-        start = time()
-        client.delete_dataset(dset.name, timeout=30)
-        client.delete_model(model.name, timeout=30)
-        deletion_time = time() - start
+            # pd raster finalization
+            pd_raster_finalization_time = time_it(
+                model_raster.finalize_inferences, dset_raster
+            )
+
+            try:
+                eval_base_raster = run_base_evaluation(
+                    dset=dset_raster, model=model_raster
+                )
+                # eval_pr_raster = run_pr_curve_evaluation(
+                #     dset=dset_raster, model=model_raster
+                # )
+                # eval_detail_raster = run_detailed_pr_curve_evaluation(
+                #     dset=dset_raster, model=model_raster
+                # )
+            except TimeoutError:
+                raise TimeoutError(
+                    f"Evaluation timed out when processing {limit} datums."
+                )
+
+            start = time()
+            client.delete_dataset(dset_raster.name, timeout=30)
+            client.delete_model(model_raster.name, timeout=30)
+            raster_deletion_time = time() - start
 
         results = {
-            "number_of_datums": limit,
-            "number_of_unique_labels": eval_.meta["labels"],
-            "number_of_annotations": eval_.meta["annotations"],
-            "ingest_runtime": f"{(ingest_time):.1f} seconds",
-            "eval_runtime": f"{(eval_.meta['duration']):.1f} seconds",
-            "del_runtime": f"{(deletion_time):.1f} seconds",
+            "box": {
```

Comment:
I love your idea of testing different annotation types! could you add back `rasters_from_multipolygon` (from the original benchmark script) and include additional runs for `polygon`, `multipolygon`, `rasters_from_bbox`, and `rasters_from_polygon`? that would give us a few really helpful comparisons to understand how postgis/numpy work on all of our supported annotation types.

another important validation we could do would be to make sure we get the same output from all of these different tests. I can look into that in a separate PR

Reply:
see above ^ the benchmark now checks all permutations of geometry type.